Removing `error` extension function (#874)
cedar-policy/cedar: cedar-policy/CHANGELOG.md

### Removed

> - Reduced precision of partial evaluation for `||`, `&&`,  and conditional expressions.

Reviewer: Maybe we want to be more specific here? An example would help.

### Removed

> - Reduced precision of partial evaluation when evaluating lazy
>   expressions

Reviewer 2: Probably makes more sense under the "changed" header than "removed" header. It also may be unclear to users what lazy expressions means in this context, so I'd suggest just
```suggestion
- Reduced precision of partial evaluation for `||`, `&&`,  and conditional expressions.
```

> - Removed the `error` extension function

Reviewer 2: ```suggestion
- Removed the `error` extension function, which was previously used during partial evaluation.
```
